### DISCUSSION TITLE
minio_s3_requests_waiting_total - waiting for what?

### DISCUSSION THREAD

**Discussion by Jazz9** [minio/minio#19845] [Q&A]
I'm seeing listobject requests waiting in promethius. The SSD drives/CPU are not very busy, what is it waiting for?

**Comment by harshavardhana** [comment]
for that we need to see a profile output of the setup.

**Reply by Jazz9** [reply]
I'm slowly getting more better at reading the minio code!

It seems the line : buffers := bpool.NewBytePoolCap(n, blockSizeV2, blockSizeV2*2) has a n value of 2048 whereas my system has 192GB ram and lots available. Tested on ubuntu 24.04 and 22.04.

I suspect func availableMemory() might not be returning the correct value for my system. I'll write a go test and see what I can find.

**Reply by Jazz9** [reply]
Available memory is reporting correctly. 

I don't see how n ever get above 2048 in func newErasureServerPools

Maybe the > should be around the other way?

	// Avoid allocating more than half of the available memory
	if maxN := availableMemory() / (blockSizeV2 * 2); n > maxN {
		n = maxN
	}

Maybe it should just be

`n := availableMemory() / (blockSizeV2 * 2)`

**Reply by Jazz9** [reply]
Making the code change to increase the cache improved the speed of the requests. Now the CPU is the bottleneck, i'll get that upgraded next week. I suspect the larger cache is more cpu intensive to manage but the performance gain with listobjects was around 50%.
